### PR TITLE
python3Packages.jenkinsapi: fix failing build

### PR DESCRIPTION
**Pull request by de11n**
## Description of changes

Build was failing when using `setuptools` build system, but with `pyproject` it works. I also refactored how tests are managed.

cc @drets

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
@ofborg build python310Packages.jenkinsapi

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 245913` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.jenkinsapi</li>
    <li>python310Packages.jenkinsapi.dist</li>
    <li>python311Packages.jenkinsapi</li>
    <li>python311Packages.jenkinsapi.dist</li>
  </ul>
</details>

**Review by natsukium — COMMENTED** [review]
Would you please add `pythonImportsCheck`?
Besides this, LGTM.

**Comment by de11n** [comment]
@natsukium Thank you. I added a `pythonImportsCheck`.

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

File: pkgs/development/python-modules/jenkinsapi/default.nix [hunk 1]
```diff
@@ -1,43 +1,39 @@
...
-  # access to macOS is not an adventure I am ready to embark on - @rski
-  doCheck = !stdenv.isDarwin;
-  # don't run tests that try to spin up jenkins, and a few more that are mysteriously broken
-  checkPhase = ''
-    py.test jenkinsapi_tests \
-      -k "not systests and not test_plugins and not test_view"
-  '';
+  nativeBuildInputs = [ flit-core pbr ];
+  propagatedBuildInputs = [ pytz requests six ];
+  nativeCheckInputs = [ pytestCheckHook mock pytest-mock ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jenkinsapi/default.nix:26]:
```suggestion
  nativeBuildInputs = [
    flit-core
    pbr
  ];

  propagatedBuildInputs = [
    pytz
    requests
    six
  ];

  nativeCheckInputs = [
    pytestCheckHook
    mock
    pytest-mock
  ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
fix is very generic, please be a bit more specific about what is being fixed

**Comment by de11n** [comment]
> fix is very generic, please be a bit more specific about what is being fixed

Updated commit message.

**Comment by natsukium** [comment]
@ofborg build pytho310Packages.jenkinsapi

**Pull request merged**